packages/library/src/react/index.ts

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+
+import { Screen, ScreenOptions } from '../html/screen'
+
+export interface ReactScreenOptions<T> extends ScreenOptions {
+  component: React.FunctionComponent<T>
+  props: T
+}
+
+/**
+ * ReactScreens have all the features of Screen as well as the
+ * ability to render a React component to a DOM element. Simply include
+ * a reference to a React.FunctionComponent in the `component` option and its props as the `props` option.
+ */
+export class ReactScreen<T> extends Screen {
+  options!: ReactScreenOptions<T>
+  componentDiv?: HTMLElement
+
+  constructor(options: ReactScreenOptions<T>) {
+    super(options)
+  }
+
+  onRun() {
+    super.onRun()
+    this.componentDiv = document.createElement('div')
+    this.options.el?.appendChild(this.componentDiv)
```

Comment:
This usage of the `el` option actually seems to have broken with the typescript update. I'm not seeing any active `el` attached to components now, even at `run` time.

Reply:
Yes, this was part of the removal of runtime data from `options`. The component `el` is now part of the `context` in which a component is presented, which in turn is added while the study is running. I think you should be able to access the specific component via `this.internals.context.el`, cf. for example the [`canvas.Screen` code](https://github.com/FelixHenninger/lab.js/blob/main/packages/library/src/canvas/screen.ts#L96-L119)

Reply:
What I ended up doing is using `global.rootEl`. Is there any meaningful difference between that and `context.el`?

Another thing I did to make this nicer in the implementation of this in our library is add `rootEl` to the `global` interface in our equivalent of `html.Screen`